shell/server/accounts/saml/saml-server.js

```diff
@@ -173,3 +175,61 @@ WebApp.connectHandlers.use(connect.urlencoded()).use(function (req, res, next) {
     middleware(req, res, next);
   }).run();
 });
+
+Meteor.methods({
+  generateSamlLogout: function () {
+    const service = generateService();
+    if (!service.logoutUrl) {
+      throw new Meteor.Error(500, "No SAML logout url specified");
+    }
+
+    const _saml = new SAML(service);
+    let identity;
+    Meteor.user().loginIdentities.forEach((_identity) => {
+      const currIdentity = Meteor.users.findOne({ _id: _identity.id, });
+      if (currIdentity.services.saml) {
+        identity = currIdentity;
+      }
+    });
+    // TODO(someday): handle user having more than one SAML identity
+
+    return Meteor.wrapAsync(_saml.getLogoutUrl.bind(_saml))({
+      user: {
+        nameID: identity.services.saml.id,
+        nameIDFormat: identity.services.saml.nameIDFormat ||
+          "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent",
+      },
+    });
+  },
+
+  validateSamlLogout: function (SAMLRequest) {
```

Comment:
type-check SAMLRequest.

Also the variable name should probably be `samlRequest`...